### PR TITLE
fix: local builds resolve dependencySizes properly

### PR DESCRIPTION
**Pull request by cgat**
When `getPackageStats` is run, the expected output includes composite information for npm dependency. This information is outputted in the `dependencySizes` array.

Previous to this commit, this information was not getting calculated correctly when `getPackageStats` was given a local path to install. The issue was that npm installs local package with a different directory structure than with a remote package.

Local:
`<installpath>/node_modules/package.json`
`<installpath>/node_modules/<packageName>` // this a symlink to the local package
`<installpath>/node_modules/<packageName>/package.json`
`<installpath>/node_modules/<packageName>/node_modules`

Remote:
`<installpath>/node_modules/package.json`
`<installpath>/node_modules/*` // All the packages (deps and the package itself) are install here.

The parsing logic in `getDirectorySizeTree` did not account for the extra `node_modules` in the module path and due to this was not calculating the dep tree correctly.

### CODE REVIEW

**Comment by cgat** [comment]
@pastelsky Any chance we can get this merged and released this weekend?

**Comment by pastelsky** [comment]
@cgat Thanks again for fixing this so quickly, have left a couple of comments – upon resolving which we should be ready to release this.

**Comment by cgat** [comment]
@pastelsky I don't see the comments? Maybe you have to request changes so I can seem them?

**Comment by pastelsky** [comment]
@cgat the comments should be visible now.

File: src/utils/build.utils.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ const BuildUtils = {
     return uniqueMissingModules
   },
 
-  async buildPackage({ name, installPath, externals, options }) {
+  async buildPackage({ name, installPath, externals, isLocal, options }) {
```
[the remark this answers is not in the document]
Comment (cgat) [reply src/utils/build.utils.js]: Yeah, I agree that passing `isLocal` around isn't the best solution, but I was concerned that I miss an edge case if I tried to test the path and break the normal functionality. I don't have the name of the package in this method (I should double check `stats` though), so the only other way I think of is to test whether `node_modules` under the install path has a single symbolic link as its only child. In that case, it would have to be a local install. Any objections to this? I can get a solution out tomorrow.

[the remark this answers is not in the document]
Comment (pastelsky) [reply src/utils/build.utils.js]: Let's pass the package name from `build.utils.js#buildPackage` to `getDependencySizeTree.js#bundleSizeTree` and remove prefixes until first occurrence.

**Comment by cgat** [comment]
@pastelsky So I've spent the morning digging into this. Regarding this:

> Let's pass the package name from build.utils.js#buildPackage to getDependencySizeTree.js#bundleSizeTree and remove prefixes until first occurrence.

The module paths don't include the package name for the non-local use case. So for example, if I was trying to run this against the `helmet` package and that package has a dependency `dasherize`, the module path would be `/private/tmp/tmp-build/packages/build-helmet-xih/node_modules/dasherize` because `npm` flattens dependencies (ie it's not located under `node_modules/helmet`). So we can't just look for the package name in the module path. 

The second thing I found was that `yarn add <localPath>` behaves differently than `npm install <localPath>`. `npm` just creates a symlink to the local directory that you give it. So If I was in a directory `/tmp/tmp-build/packages/test` and ran `npm install /path/to/helmet`, I would end up with:
`/tmp/tmp-build/packages/test/node_modules/helmet //symlink to /path/to/helmet`. While `yarn` actually install the package as if it were remote. More specifically, the `node_modules` would look have flatten dependencies and would look the same as if you run `yarn install helmet`.

The take away is that the real issue is how `npm` handles the installation of local folders (just symlinking). However, there is a workaround: `npm install $(npm pack <folder> | tail -1)` will package the folder (installing the dependencies as needed) and behave like `yarn add <folder>`. This has the added benefit of working with monorepos where `node_modules` is not located with the package but at the root. If we went with this approach, I'd pass in `isLocal` to `InstallationUtils.installPackage` and wrap the package string with the pack command above.

Alternatively, we could leave it up to the user to doing the packing, and instead support packaged files.

Thoughts? Let me know if you need any clarification.

**Comment by cgat** [comment]
@pastelsky I've opened a separate PR with the packing method: https://github.com/pastelsky/package-build-stats/pull/16

I really believe the `pack` method is the optimal approach, but if you want me to continue with the above approach, I can update this PR as well.

FYI, It would reaaaally help me if we could close this out and release before the end of week. I'm being resourced onto a different project soon and have an open PR that is blocked by this. With that being said, if you are overworked and don't have the time, I fully understand. Just trying to give you some context.

**Comment by cgat** [comment]
Just an FYI, I'm on holidays for the next two weeks in case I don't respond.

**Comment by cgat** [comment]
@pastelsky ...?

**Comment by cgat** [comment]
@pastelsky Just pinging you again. I did end up publishing a fork so I could use this via npm: https://www.npmjs.com/package/package-build-stats-local, for those of you who are interested.

**Comment by pastelsky** [comment]
I like the npm pack approach. Keeps things similar to remote packages. I've merged that PR, and hence declining this one.

Its been long since you raised it – apologies for the long delay. I'm glad that feature can be used in future for many more applications.

**Comment by cgat** [comment]
@pastelsky Thanks 🙏 . Please let me know when you get the chance to make another release.

**Comment by pastelsky** [comment]
@cgat Published as `6.2.0`
